## This notebook represents how I update my learning curve
**A 100 Day Challenge where I turn medium articles on machine learning, data science, data analyis...in to Notebooks**

**1**

### Descriptive statistics is a study of data analysis to describe, show or summarize data in a meaningful way. It involves the calculation of various measures such as the measure of center, the measure of variability, percentiles and also the construction of tables & graphs.

In [1]:
# Prerequisites Python libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from collections import defaultdict # Defaultdict is a sub-class of the dict class 
import thinkplot

**Categorical data: non-numerical information such as gender, race, religion, marital status etc, it can also be ordinal, ranks in numbers etc**


**Numerical data: measurement or count such as height, weight, age, salary, number of children, etc**

In [2]:
# Let me read the melbourne dataset
melHouseData = pd.read_csv('MELBOURNE_HOUSE_PRICES_LESS.csv')
melHouseData.head()

Suburb           Address  Rooms Type      Price Method   SellerG  \
0    Abbotsford     49 Lithgow St      3    h  1490000.0      S    Jellis   
1    Abbotsford     59A Turner St      3    h  1220000.0      S  Marshall   
2    Abbotsford     119B Yarra St      3    h  1420000.0      S    Nelson   
3    Aberfeldie        68 Vida St      3    h  1515000.0      S     Barry   
4  Airport West  92 Clydesdale Rd      2    h   670000.0      S    Nelson   

        Date  Postcode             Regionname  Propertycount  Distance  \
0  1/04/2017      3067  Northern Metropolitan           4019       3.0   
1  1/04/2017      3067  Northern Metropolitan           4019       3.0   
2  1/04/2017      3067  Northern Metropolitan           4019       3.0   
3  1/04/2017      3040   Western Metropolitan           1543       7.5   
4  1/04/2017      3042   Western Metropolitan           3464      10.4   

                  CouncilArea  
0          Yarra City Council  
1          Yarra City Council  
2          Yarra City Council  
3  Moonee Valley City Council  
4  Moonee Valley City Council

In [3]:
# To get further details of the data type for each variable in our dataset
melHouseData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 63023 entries, 0 to 63022
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         63023 non-null  object 
 1   Address        63023 non-null  object 
 2   Rooms          63023 non-null  int64  
 3   Type           63023 non-null  object 
 4   Price          48433 non-null  float64
 5   Method         63023 non-null  object 
 6   SellerG        63023 non-null  object 
 7   Date           63023 non-null  object 
 8   Postcode       63023 non-null  int64  
 9   Regionname     63023 non-null  object 
 10  Propertycount  63023 non-null  int64  
 11  Distance       63023 non-null  float64
 12  CouncilArea    63023 non-null  object 
dtypes: float64(2), int64(3), object(8)
memory usage: 6.3+ MB


**The result reveals a total of 63023 entries in the dataset. Generally, the columns tied with data type 
“int64” and “float64” denotes numerical data while data type “object” denotes categorical data.**

*This is not entirely through, some numerical int64 data types may represent ranks like class e.g. poor, rich, middle-class*

*The only exception is “Postcode”. Although the postcode is represented as a number (int64),
it doesn’t necessarily make it quantitative. Postcode is just numbers applied to categorical data.*

**Here is another contradiction**

### Measure of Center

**One common way to summarize our numerical data is to find out the central tendency of our data. 
For example, we might question “What is the most typical value of the house price in our dataset?”. 
To address this question, we can resort to the two most common measures of center: mean and median.**

**Mean is an average of all the numbers. The steps required to calculate a mean are:
sum up all the values of a target variable in the datasetthe sum by the number of values**

*Note of warning: a mean might be influenced by the outliers.*

**Outliers are the numbers which are either extremely high or extremely low compared to the rest of the numbers in a dataset.**

**Median is the middle value of a sorted list of numbers. The steps required to get a median from a list 
of numbers are:**

**Sort the numbers from smallest to highest**

**if the list has an odd number of values, the value in the middle position is the median**

**if the list has an even number of values, the average of the two values in the middle will be the median**


*Note: A median is not influenced by the outliers.*

In [4]:
# Calculating the mean of the 'Price' column
average = melHouseData['Price'].mean()
print(average)

997898.2414882415


In [5]:
median = melHouseData['Price'].median()
print(median)

830000.0


### Measure of Variation

*This definition is not from this medium article*

**Variability (also called spread or dispersion) refers to how spread out a set of data is. Variability gives you a way to describe how much data sets vary and allows you to use statistics to compare your data to other sets of data. The four main ways to describe variability in a data set are:**

*Range*

*Interquartile range*

*Variance*

*Standard deviation.*

*The Definition below is from the medium article*

**Standard Deviation: One common method to measure the variation of our dataset is to calculate the standard deviation (SD). The SD is just a measurement to tell how a set of values spread out from their mean. A low SD shows that the values are close to the mean and a high SD shows a high diversion from the mean.**

**The steps to calculate SD are as follows:**

*1.Calculate the mean of a dataset*

*2.For each number in the dataset, subtract it with the mean*

*3.Square the difference obtained from Step 2*

*4.Sum up the results from Step 3*

*5.Divide the summation from Step 4 by the number of values in the dataset minus one*

*6.Square root the result from Step 5*

*Note:*

**1. SD must be a positive number**

**2.SD is affected by outliers as its calculation is based on the mean**

**3.The smallest possible value of SD is zero. If SD is zero, all the numbers in a dataset share the same value.**

In [6]:
# Pandas also offer a handy method, std(), to calculate the SD
standard_deviation = melHouseData['Price'].std()

In [7]:
print("Stardard Deviation:", standard_deviation)

Stardard Deviation: 593498.9190372769


## Graphical way to show the numerical data distribution